src/platform/windows/drop_handler.rs

```diff
@@ -173,12 +181,14 @@ impl FileDropHandler {
                 }
             }
 
-            return hdrop;
+            return Some(hdrop);
+        } else if get_data_result == DV_E_FORMATETC {
+            // If the dropped item is not a file this error will occur.
+            // In this case it is OK to return without taking further action.
+            return None;
+        } else {
+            panic!("Unexpected error occured while processing dropped item");
```

Comment:
Using a panic to unwind into foreign code is undefined behavior, so I don't think panicking is the best move here. Would it be better to ignore any other errors? Another option would be to catch panics and propagate them to the parent thread when they happen, like in #703.

Reply:
I really like the idea of using log instead as it could be the case that the code is not prepared for some specific scenario where it is best to just indicate that the drop event is not supported instead of panicking.